aten/src/ATen/core/function_schema.h

```diff
@@ -373,6 +451,61 @@ struct FunctionSchema {
 
     return is_mutable(*index);
   }
+  bool areAliasing(const SchemaArgument& lhs, const SchemaArgument& rhs) const {
+    TORCH_INTERNAL_ASSERT(
+        (lhs.index < getCorrectList(lhs.type).size()),
+        "Invalid index for schema.");
+    TORCH_INTERNAL_ASSERT(
+        (rhs.index < getCorrectList(rhs.type).size()),
+        "Invalid index for schema.");
+
+    const Argument lhsArg = getCorrectList(lhs.type)[lhs.index];
+    const Argument rhsArg = getCorrectList(rhs.type)[rhs.index];
+    if (lhsArg.alias_info() && rhsArg.alias_info()) {
+      for (const auto& lhsSet : lhsArg.alias_info()->afterSets()) {
+        for (const auto& rhsSet : rhsArg.alias_info()->afterSets()) {
+          if (lhsSet == rhsSet) {
+            return true;
+          }
+        }
+      }
+    }
+    return false;
+  }
+  bool areAliasing(const SchemaArgument& lhs, const SchemaArgument& rhs, bool check_additional) const {
+    bool basic_check = areAliasing(lhs, rhs);
+    if (check_additional) {
+      const c10::Argument lhsArg = getCorrectList(lhs.type)[lhs.index];
+      const c10::Argument rhsArg = getCorrectList(rhs.type)[rhs.index];
+      if (lhsArg.alias_info() && lhsArg.alias_info()->isWildcardAfter()) {
+        c10::optional<std::vector<TypePtr>> lhsTypes = mapTypeToAliasTypeSet(lhsArg.type());
+        c10::optional<std::vector<TypePtr>> rhsTypes = mapTypeToAliasTypeSet(lhsArg.type());
+        if (lhsTypes && rhsTypes) {
+          std::unordered_set<TypePtr> flattened_types;
+          flattenAliasTypeSet(ArrayRef<TypePtr>(*lhsTypes), flattened_types);
+          if (std::all_of((*rhsTypes).begin(), (*rhsTypes).end(), [&flattened_types](const TypePtr& type) {
+            return flattened_types.count(type);
+          })) {
+            return true;
+          }
+        }
+      }
+      if (rhsArg.alias_info() && rhsArg.alias_info()->isWildcardAfter()) {
```

Comment:
could we put this into a lambda and then call the lambda twice with (lhsArg, rhsArg) and (rhsArg, lhsArg) to reduce code duplication?